src/binder/clrprivbindercoreclr.cpp

```diff
@@ -157,10 +149,12 @@ HRESULT CLRPrivBinderCoreCLR::BindUsingPEImage( /* in */ PEImage *pPEImage,
                 hr = BindAssemblyByNameWorker(pAssemblyName, &pCoreCLRFoundAssembly, true /* excludeAppPaths */);
                 if (SUCCEEDED(hr))
                 {
+                    // Is identity same as TPA assembly
                     if (pCoreCLRFoundAssembly->GetIsInGAC())
                     {
-                        // If we were able to bind to a TPA assembly, then fail the load
-                        IF_FAIL_GO(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+                        // Fail bind if requested assembly is at different location than TPA assembly
+                        if(pPEImage != pCoreCLRFoundAssembly->GetPEImage())
```

Comment:
Yes this is correct if you do loadfrom for both the assemblies. I believe the behavior that we have currently is fine for LoadFrom because there are explicit calls to do loadfrom with different path to same assembly. The user seems to be just playing around here. However, for TPA assemblies these can conflict can happen between Assembly.Load & Assembly.LoadFrom. User does not have total idea what files are in TPA and so if a conflict does happen it is better to throw an exception instead of returning an assembly from TPA.


I am trying to be explicit and throwing an exception instead of returning or loading assembly at different location.